### ISSUE TITLE
Failed Build on OSX (El Capitan)

### ISSUE THREAD

**Issue by ninosvo** [OpenVPN/openvpn3#35]
**I had 20+ failed build errors similar to other posts. When I added ASIO=1 it brings it down to 1 error:**

----
MTLS=1 LZ4=1 ASIO=1 build cli
In file included from cli.cpp:58:
In file included from /Users/XYZ/src/ovpn3/core/client/ovpncli.cpp:78:
/Users/XYZ/src/ovpn3/core/openvpn/log/logthread.hpp:122:20: error: thread-local storage is not supported for the current target
    OPENVPN_EXTERN thread_local OPENVPN_LOG_CLASS* global_log; // GLOBAL
                   ^
1 error generated.
---

**If I add the ASIO_DIR it is slightly different error:**

---
MTLS=1 LZ4=1 ASIO=1 ASIO_DIR=~/src/mac build cli
In file included from cli.cpp:58:
In file included from /Users/XYZ/src/ovpn3/core/client/ovpncli.cpp:30:
/Users/XYZ/src/ovpn3/core/openvpn/io/io.hpp:28:10: fatal error: 'asio.hpp' file not found
#include <asio.hpp>
         ^
1 error generated.
---

Help please?

**Comment by lstipakov** [comment]
Hello,

I believe you don't need to specify location if ASIO if you use following build command inside $O3/core/scripts/mac directory:

`➜  mac git:(master) ✗ O3=~/Projects/ovpn3 DEP_DIR=$O3/deps DL=$O3/dl OSX_ONLY=1 ./build-all`

Regarding thread-local problem - that looks odd. Which clang you use?

```
➜  mac git:(master) ✗ clang --version
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin17.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Could you try with latest homebrew's clang instead of Apple's?

**Comment by ninosvo** [comment]
My clang version:

`Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
`
I installed another one via brew

`
brew install llvm
`

Then adding it to the path together with the build command worked with the following warnings

`
PATH="/usr/local/opt/llvm/bin:$PATH" MTLS=1 LZ4=1 ASIO=1 build cli

...
/Users/nino/src/mac/asio/asio/include/asio/impl/write_at.hpp:552:51: warning: unused typedef 'type_check' [-Wunused-local-typedef]
  ASIO_WRITE_HANDLER_CHECK(WriteHandler, handler) type_check;
                                                  ^
69 warnings generated.
`

Seems to have built. I tested an ovpn config I use with tunnelblick and failed to connect. 

`
Wed Feb 14 20:17:55.855 2018 Wed Feb 14 20:17:55.855 2018 TUN Error: utun_error: cannot open available utun device
Wed Feb 14 20:17:55.855 2018 EVENT: TUN_IFACE_CREATE utun_error: cannot open available utun device [FATAL-ERR]
Wed Feb 14 20:17:55.855 2018 EVENT: DISCONNECTED
Wed Feb 14 20:17:55.855 2018 Client exception in transport_recv: tun_exception: not connected
`

Thanks for the help with the build. I'll keep digging to figure out the connection.

**Comment by varun-ll** [comment]
> My clang version:
> 
> `Apple LLVM version 7.3.0 (clang-703.0.31) Target: x86_64-apple-darwin15.6.0 Thread model: posix InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin `
> 
> I installed another one via brew
> 
> `brew install llvm`
> 
> Then adding it to the path together with the build command worked with the following warnings
> 
> `PATH="/usr/local/opt/llvm/bin:$PATH" MTLS=1 LZ4=1 ASIO=1 build cli` `/Users/nino/src/mac/asio/asio/include/asio/impl/write_at.hpp:552:51: warning: unused typedef 'type_check' [-Wunused-local-typedef] ASIO_WRITE_HANDLER_CHECK(WriteHandler, handler) type_check; ^ 69 warnings generated.`
> 
> Seems to have built. I tested an ovpn config I use with tunnelblick and failed to connect.
> 
> `Wed Feb 14 20:17:55.855 2018 Wed Feb 14 20:17:55.855 2018 TUN Error: utun_error: cannot open available utun device Wed Feb 14 20:17:55.855 2018 EVENT: TUN_IFACE_CREATE utun_error: cannot open available utun device [FATAL-ERR] Wed Feb 14 20:17:55.855 2018 EVENT: DISCONNECTED Wed Feb 14 20:17:55.855 2018 Client exception in transport_recv: tun_exception: not connected`
> 
> Thanks for the help with the build. I'll keep digging to figure out the connection.

Getting the same error. Any luck?
